packages/compass-workspaces/src/components/workspaces.tsx

```diff
@@ -0,0 +1,186 @@
+import { connect } from 'react-redux';
+import { AppRegistryProvider } from 'hadron-app-registry';
+import { WorkspaceTabs, css } from '@mongodb-js/compass-components';
+import type {
+  OpenWorkspaceOptions,
+  WorkspaceByType,
+  WorkspaceTab,
+  WorkspacesState,
+} from '../stores/workspaces';
+import {
+  closeTab,
+  getActiveTab,
+  getLocalAppRegistryForTab,
+  moveTab,
+  emitOnTabChange,
+  openTabFromCurrent,
+  selectNextTab,
+  selectPrevTab,
+  selectTab,
+} from '../stores/workspaces';
+import { useWorkspacePlugin } from './workspaces-provider';
+import toNS from 'mongodb-ns';
+
+const workspacesContainerStyles = css({
+  display: 'grid',
+  width: '100%',
+  height: '100%',
+  gridTemplateColumns: '100%',
+  gridTemplateRows: 'auto 1fr',
+});
+
+const workspacesContentStyles = css({
+  display: 'flex',
+  flex: 1,
+  minHeight: 0,
+});
+
+type CompassWorkspacesProps = {
+  tabs: WorkspaceTab[];
+  activeTab?: WorkspaceTab | null;
+  initialTab?: OpenWorkspaceOptions;
+
+  onSelectTab(at: number): void;
+  onSelectNextTab(): void;
+  onSelectPrevTab(): void;
+  onMoveTab(from: number, to: number): void;
+  onCreateTab(): void;
+  onCloseTab(at: number): void;
+
+  onTabChange?: (tab: WorkspaceTab) => void;
+};
+
+const CompassWorkspaces: React.FunctionComponent<CompassWorkspacesProps> = ({
+  tabs,
+  activeTab,
+  onSelectTab,
+  onSelectNextTab,
+  onSelectPrevTab,
+  onMoveTab,
+  onCreateTab,
+  onCloseTab,
+  onTabChange,
+}) => {
+  const Workspace = useWorkspacePlugin(activeTab?.type ?? '');
+
+  useEffect(() => {
+    if (activeTab) {
+      onTabChange?.(activeTab);
+    }
+  }, [
+    onTabChange,
+    activeTab?.type,
+    (activeTab as { namespace: string } | undefined)?.namespace,
+  ]);
+
+  const tabDescriptions = useMemo(() => {
+    return tabs.map((tab) => {
+      switch (tab.type) {
+        case 'My Queries':
+          return {
+            id: tab.id,
+            title: tab.type,
+            iconGlyph: 'CurlyBraces',
+          };
+        case 'Databases':
+          return {
+            id: tab.id,
+            title: tab.type,
+            iconGlyph: 'Database',
+          };
+        case 'Performance':
+          return {
+            id: tab.id,
+            title: tab.type,
+            iconGlyph: 'Gauge',
+          };
+        case 'Collections':
+          return {
+            id: tab.id,
+            title: tab.namespace,
+            subtitle: tab.type,
+            iconGlyph: 'Database',
+          };
+        case 'Collection': {
+          const { database, collection } = toNS(tab.namespace);
+          return {
+            id: tab.id,
+            title: collection,
+            subtitle: `${database} > ${collection}`,
+            iconGlyph:
+              tab.collectionType === 'view'
+                ? 'Visibility'
+                : tab.collectionType === 'timeseries'
+                ? 'TimeSeries'
+                : 'Folder',
+          };
```

Comment:
Could we somehow attach this to the tab definitions themselves instead (in addition to `name`/`component`) instead of maintaining this switch here?

Reply:
Yes, totally! Now that they are all used in one place it should be easy enough, I'll do another pass here and change the type dependency to make it so that it's coming from the plugins exporting the tabs 👍

Reply:
Done in https://github.com/mongodb-js/compass/pull/5169/commits/1ec06fb790f0af2daec50ff4ba3b364a763f07d3